### PR TITLE
[FIX] 🐛 (#288) - avoid Page Reload `onClick` charity About btn 

### LINKED ISSUE #288: [BUG] `About` button under Select Charity causes page reload :(
## BUG:
While creating new farm, `About` button under Select Charity reloads the page

## SCOPE:
UX

## EFFECTS:
Need to re-sign the request in metamask :(

## CAUSES:
- Default event is fired for button `onClick`
- Missing `target` value in `window.open()` _(should be passed for better for UX)_
https://github.com/sunflower-land/sunflower-land/blob/64d84e268a255b638d5972b5448827d1e8278f9a/src/features/auth/components/Donation.tsx#L62

## FIX :)
- Passing `target="_bank"` will open link in new tab _(better UX)_
```diff
const onAboutClick = (url: string) => {
-    window.open(url);
+    window.open(url, "_blank");
  };
```

- Prevent **default** behavior for button onClick evt
```tsx
<Button
  ...
  onClick={(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    onAboutClick(url), e.preventDefault();
  }}
  ...
>
```

- Modifying `Button` RFC, define std mouse event type for `onClick`
```tsx
interface Props {
  onClick?:
      | ((event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)
      | (() => void);
  ...
}
```

### ~Opening~ Opened PR #289 with Fix :)

### PR DESCRIPTION
**Pull request by Pranav-yadav**
# Description
This PR should fix 🐛  #288 
- which was causing page reload on clicking About button under charity option, while user is trying to create new farm

Fixes #288

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
- `yarn test`
- Manual Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/ui/Button.tsx
Comment (adamhannigan) [reply src/components/ui/Button.tsx:6]: I would suggest removing the second option `(() => void)` since every `onClick` will have the `event` parameter passed through.

Parameters are optional to be implemented in the parent components so they do not need to use the `event` parameter, they can leave it blank :)

**Review by adamhannigan — APPROVED** [review]
Good find 👌 

I would suggest only keeping the `(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void)` definition as this is the true definition of the function signature.

File: src/components/ui/Button.tsx
Comment (Pranav-yadav) [reply src/components/ui/Button.tsx:6]: Yep. Just wanted to be sure :) as Button is widely used.
PS: Tho I've tested manually every button, all r working fine no side effects

**Comment by Pranav-yadav** [comment]
PS: Don't forget to link #289 before merging 😄

**Comment by jcads** [comment]
> PS: Don't forget to link #288 before merging 😄

To automatically link the issue you need to do `Fixes #288`, without the parentheses.

**Comment by Pranav-yadav** [comment]
> > PS: Don't forget to link #288 before merging 😄
> 
> To automatically link the issue I think you need to do `Fixes #288`, without the parentheses.

@jcads afaik, issues can be only linked manually by repo maintainer/owner. Tho we can mention then anywhere in the repo discussion's. :)

**Pull request merged**
